### PR TITLE
Add Versions class; support for version filter

### PR DESCRIPTION
**Pull request by dinjas**
Addresses #58

### CODE REVIEW

File: lib/paper_trail_scrapbook/versions.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      else
+        []
+      end
+    end
+
+    def filtered(object_versions)
+      if object.respond_to?(:version_filter)
+        object_versions.select { |v| object.version_filter(v) }
+      else
+        object_versions
```
Comment (tjchambers) [inline lib/paper_trail_scrapbook/versions.rb:23]: orobably would change this to a guard clause construction

File: lib/paper_trail_scrapbook/versions.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+module PaperTrailScrapbook
+  class Versions
+    include Concord.new(:object)
+
+    def versions
+      filtered(object.versions | related_content).sort_by(&:created_at)
+    end
+
+    def related_content
+      if object.respond_to?(:trailed_related_content)
```
Comment (tjchambers) [inline lib/paper_trail_scrapbook/versions.rb:12]: change to guard claus construction?

File: spec/paper_trail_scrapbook/versions_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+module PaperTrailScrapbook
+  ::RSpec.describe Versions do
+    before do
+      PaperTrailScrapbook.config.whodunnit_class = Person
+      PaperTrail.request.whodunnit = person.id
+    end
+
+    let(:book) do
+      b = Book.new(title: 'How the Grinch stole Xmas')
```
Comment (tjchambers) [inline spec/paper_trail_scrapbook/versions_spec.rb:14]: clean with `tap`

File: spec/paper_trail_scrapbook/versions_spec.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      b.save!
+      b
+    end
+    let(:person) { Person.create(name: 'The Tim Man') }
+    let!(:authorship) do
+      a = Authorship.new(book: book, author: person)
```
Comment (tjchambers) [inline spec/paper_trail_scrapbook/versions_spec.rb:20]: clean with tap?

**Review by tjchambers — APPROVED** [review]

**Comment by tjchambers** [comment]
I approved, but if I had one addition it would be a short addition to the readme as to how to use the version_filter.

**Comment by dinjas** [comment]
Good idea. Will add that this weekend sometime.

**Pull request merged**